Plugins/GitUIPluginInterfaces/GitRevision.cs

```diff
@@ -19,7 +19,7 @@ public sealed class GitRevision : IGitItem, INotifyPropertyChanged
         public const string CombinedDiffGuid = "3333333333333333333333333333333333333333";
 
         public static readonly Regex Sha1HashRegex = new(@"^[a-f\d]{40}$", RegexOptions.Compiled);
-        public static readonly Regex Sha1HashShortRegex = new(@"\b[a-f\d]{7,40}\b", RegexOptions.Compiled);
+        public static readonly Regex Sha1HashShortRegex = new(@"\b[a-f\d]{7,40}\b(?![^@ \t\r\n]*@)", RegexOptions.Compiled);
```

Comment:
Same change should be applied to `Sha1HashRegex` (there is not even a check for boundary after 40 chars there).
the regexes should be merged too (result to check the length).
 ```suggestion
        public static readonly Regex Sha1HashShortRegex = new(@"\b[a-f\d]{7,40}\b(?![^@\s]*@)", RegexOptions.Compiled);
```

Reply:
I took over the simplification `\s`.

> Same change should be applied to Sha1HashRegex (there is not even a check for boundary after 40 chars there).

Though `Sha1HashRegex` checks the entire string already: `^...$`.